src/frontend/src/common/queries/auth.ts

```diff
@@ -75,12 +71,10 @@ export function useUpdateUserInfo(): UseMutationResult<
 
   return useMutation(updateUserInfo, {
     onSuccess: async () => {
-      await queryClient.invalidateQueries([
-        USE_USER_INFO,
-        USE_GROUP_INFO,
-        USE_GROUP_INVITATION_INFO,
-        USE_GROUP_MEMBER_INFO,
-      ]);
+      // invalidateQueries only works for multiple queries if they start with the same key
+      // https://github.com/TanStack/query/discussions/2057#discussioncomment-549333
+      await queryClient.invalidateQueries([USE_USER_INFO]);
+      await queryClient.invalidateQueries([GROUP_QUERY_PREFIX]);
```

Comment:
I like this, but it seems wacky to me that react-query invalidates all queries that match based on prefix. I was assuming the invalidation was based on string `===` equality, but it's... prefix matching? Okay, yeah, I see it in the docs now. What a wacky API. Seems like this is the way to go, then!

Reply:
ya, string equality seems much more straightforward, but here we are 🤷‍♀️